# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1331501289 25200
#      Sun Mar 11 14:28:09 2012 -0700
# Node ID c57f9c911053a0597c630f0bf4e6fecded3b864d
# Parent  2232931f24947fa56225866d2424ec29de1567c9
regexp.cc: Remove redundant condition in if statement

regexp.cc (match): Remove redundant condition in if statement

diff --git a/liboctave/regexp.cc b/liboctave/regexp.cc
--- a/liboctave/regexp.cc
+++ b/liboctave/regexp.cc
@@ -328,18 +328,17 @@ regexp::match (const std::string& buffer
         {
           int pos_match = 0;
           Matrix token_extents (matches-1, 2);
 
           for (int i = 1; i < matches; i++)
             {
               if (ovector[2*i] >= 0 && ovector[2*i+1] > 0
                   && (i == 1 || ovector[2*i] != ovector[2*i-2]
-                      || ovector[2*i-1] != ovector[2*i+1])
-                  && ovector[2*i] >= 0 && ovector[2*i+1] > 0)
+                      || ovector[2*i-1] != ovector[2*i+1]))
                 {
                   token_extents(pos_match,0) = double (ovector[2*i]+1);
                   token_extents(pos_match++,1) = double (ovector[2*i+1]);
                 }
             }
 
           token_extents.resize (pos_match, 2);
 
